### DISCUSSION TITLE
How can I perform a DFU from an ESP32 to a nRF52840?

### DISCUSSION THREAD

**Discussion by hslourenc** [zephyrproject-rtos/zephyr#53113] [Q&A]
I have a nRF52840 and an ESP32 and I want to perform a DFU to nRF52840 from ESP32, supposing I have the signed binary file in the ESP32. I was already able to perform the DFU to nRF52840 using MCUboot, making nRF52840 enter recovery mode and using a mcumgr client on my laptop to run `mcumgr -t 20 --conntype serial --connstring=COM6 image upload app_update.bin`, but I don't know how can I perform the DFU from the ESP32, can anyone help me with this? I am using Zephyr v3.1.99-ncs1 in nRF52840 and ESP-SDK in ESP32.

**Comment by rftafas** [comment]
Hi!

[There was a similar question here](https://github.com/mcu-tools/mcuboot/issues/1181). So, you would need mcumgr server on nRF ([seems available](https://github.com/zephyrproject-rtos/mcumgr)) but also port the client to ESP-IDF/FreeRTOS (not available).

If you are using ESP32 as a connectivity device, [this link may help](https://devzone.nordicsemi.com/f/nordic-q-a/86027/dfu-approach-from-cloud-to-esp32-to-nrf52840).

Probably there is code to be written on your end. I can say about how hard it will be.

**Reply by hslourenc** [reply]
Hi!

Thank you for your answer, from the researches I've been doing, I guess that this would be the solution indeed.

**Reply by rkyurpan256** [reply]
Hello, @hslourenc did you manage to find solution to your question? I on the same path there...
